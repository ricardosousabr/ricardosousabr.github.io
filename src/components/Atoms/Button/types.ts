import React, { ReactNode } from "react";

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  href?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement> | undefined
  type?: "submit" | "reset" | "button";
  background: string;
  padding: string;
  borderRadius: string;
  color: string;
  border: string;
}
