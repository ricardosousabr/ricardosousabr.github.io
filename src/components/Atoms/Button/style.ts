import styled from "styled-components";
import { ButtonProps } from "./types";

export const ButtonStyle = styled.button<ButtonProps>`
  background: ${(p) => p.background};
  padding: ${(p) => p.padding};
  border-radius: ${(p) => p.borderRadius};
  color: ${(p) => p.color};
  border: ${(p) => p.border};
  font-size: ${(p) => p.fontSize};
  font-family: 'Roboto', sans-serif;
  text-decoration: ${(p) => p.textDecoration};
`;
