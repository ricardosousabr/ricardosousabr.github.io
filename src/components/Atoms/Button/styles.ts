import styled from "styled-components";
import { ButtonProps } from "./types";

export const ButtonStyle = styled.button<ButtonProps>`
  background: transparent;
  padding: 8px 16px;
  color: white;
  border: 1px solid #bb74ce;
  font-size: 16px;
  font-family: 'Fira Code', monospace;
  text-decoration: none ;

  &:hover {
    cursor: pointer;
  }
`;
