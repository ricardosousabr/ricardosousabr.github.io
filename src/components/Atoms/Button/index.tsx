import { ButtonProps } from "./types";
import { ButtonStyle } from "./style";

export default function Button({ children, type, onClick, href, background, padding, borderRadius, color, border, fontSize, textDecoration }: ButtonProps) {
  const buttonType = {
    type: type,
    onClick: onClick,
  };
  const anchorType = {
    as: "a",
    href: href,
    textDecoration: textDecoration,
  };

  return (
    <ButtonStyle {...(href ? anchorType : buttonType)} background={background} padding={padding} borderRadius={borderRadius} color={color} border={border} textDecoration={textDecoration} fontSize={fontSize} >{children}</ButtonStyle>
  );
}
