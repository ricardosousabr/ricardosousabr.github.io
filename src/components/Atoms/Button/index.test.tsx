import renderer from "react-test-renderer";
import Button from "./index";

test("Link renders correctly", () => {
  const tree = renderer.create(<Button>Click</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
