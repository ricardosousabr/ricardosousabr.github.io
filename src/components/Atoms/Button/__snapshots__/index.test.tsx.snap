// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<button
  background="white"
  border="none"
  borderRadius="8px"
  className="sc-bdnyFh kXSFtr"
  color="black"
  padding="8px 35px"
  type="button"
>
  Click
</button>
`;
